backend/compilers/download.py

```diff
@@ -237,6 +238,133 @@ def get_compiler(
     return True
 
 
+def get_library(
```

Comment:
I feel like the library downloading should be a separate script from the compilers downloading. Dunno what other people's opinions are though.

Reply:
ah, yeah that didn't really register to me, but I agree

Reply:
Yeah that's fair. It used to reuse a lot of the infrastructure from compilers/download.py so it made sense to have a single script, but now that it does its own stuff, it can move to its own script.

Reply:
Fixed. I also updated the docs to explain what the new libraries folder does.